components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FxaWebChannelFeature.kt

```diff
@@ -248,19 +249,33 @@ class FxaWebChannelFeature(
             return status
         }
 
+        fun JSONArray.toStringList(): List<String> {
+            val result = mutableListOf<String>()
+            for (i in 0 until this.length()) {
+                result.add(this.getString(i))
+            }
+            return result
+        }
+
         /**
          * Handles the [COMMAND_OAUTH_LOGIN] event from the web-channel.
          */
         private fun processOauthLoginCommand(accountManager: FxaAccountManager, payload: JSONObject): JSONObject? {
             val authType: AuthType
             val code: String
             val state: String
+            val declinedEngines: List<String>?
 
             try {
                 val data = payload.getJSONObject("data")
                 authType = data.getString("action").toAuthType()
                 code = data.getString("code")
                 state = data.getString("state")
+                declinedEngines = try {
+                    data.getJSONArray("declinedSyncEngines").toStringList()
```

Comment:
This block could just be: `declinedEngines = data.optJSONArray("declinedEngines")?.toStringList()` ?